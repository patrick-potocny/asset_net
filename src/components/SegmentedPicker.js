import React from "react";

function SegmentedPicker() {

  return (
    <div className="segmented-control">
    </div>
  );
}

export default SegmentedPicker;
