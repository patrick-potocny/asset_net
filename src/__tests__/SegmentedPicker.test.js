import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import SegmentedPicker from "../components/SegmentedPicker";
import "@testing-library/jest-dom/extend-expect"

describe('SegmentedPicker component', () => {
  test('should properly render SegmentedPicker compared to snapshot', () => {
    const mockCallback = jest.fn();
    const controlRef = { current: null };
    const options = [
      { value: "option1", label: "Option 1", ref: { current: null } },
      { value: "option2", label: "Option 2", ref: { current: null } },
      { value: "option3", label: "Option 3", ref: { current: null } }
    ];
    

    const { container } = render(
      <SegmentedPicker
        name="test-picker"
        options={options}
        callback={mockCallback}
        controlRef={controlRef}
        defaultOption='option1'
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("should call callback function with correct value and index on input change", () => {
    const mockCallback = jest.fn();
    const controlRef = { current: null };
    const options = [
      { value: "option1", label: "Option 1", ref: { current: null } },
      { value: "option2", label: "Option 2", ref: { current: null } },
      { value: "option3", label: "Option 3", ref: { current: null } }
    ];
    render(
      <SegmentedPicker
        name="test-picker"
        options={options}
        callback={mockCallback}
        controlRef={controlRef}
        defaultOption='option1'
      />
    );
  
    fireEvent.click(screen.getByLabelText("Option 2"));
  
    expect(mockCallback).toHaveBeenCalledWith("option2", 1);
  });
});
