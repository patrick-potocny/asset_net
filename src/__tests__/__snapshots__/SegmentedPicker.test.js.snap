// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedPicker component should properly render SegmentedPicker compared to snapshot 1`] = `
<div>
  <div
    class="container controls-container"
    style="--highlight-width: 0px; --highlight-x-pos: 0px;"
  >
    <div
      class="controls idle"
    >
      <div
        class="segment active"
      >
        <input
          checked=""
          id="Option 1"
          name="test-picker"
          type="radio"
          value="option1"
        />
        <label
          for="Option 1"
        >
          Option 1
        </label>
      </div>
      <div
        class="segment inactive"
      >
        <input
          id="Option 2"
          name="test-picker"
          type="radio"
          value="option2"
        />
        <label
          for="Option 2"
        >
          Option 2
        </label>
      </div>
      <div
        class="segment inactive"
      >
        <input
          id="Option 3"
          name="test-picker"
          type="radio"
          value="option3"
        />
        <label
          for="Option 3"
        >
          Option 3
        </label>
      </div>
    </div>
  </div>
</div>
`;
